ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAP-92 Add deletion page and shared flash banner

## Description of change
Adds the ability for user to check and then, delete an application, or cancel deleting an application.

Also adds a flash message a partial

## Link to relevant ticket
[CRIMAP-92](https://dsdmoj.atlassian.net/browse/CRIMAP-92)

## Notes for reviewer
- Given that applications don't have a state at the moment, and cannot be "finished" as the steps aren't there, the deletion does not check that they are "in progress"
- when we can check that applications are in progress this should be added to the view logic so that it doesn't render a delete button as per the designs, and similarly the delete request should error if the application is not on progress.
- We can add these checks as we are able to.

## Screenshots of changes (if applicable)

### After changes:

Confirmation page:
<img width="992" alt="Screenshot 2022-08-18 at 15 00 16" src="https://user-images.githubusercontent.com/13377553/185413889-54fa96dd-09da-45ef-94f5-3fc15e83a8e3.png">

Flash message on deletion:
<img width="1002" alt="Screenshot 2022-08-18 at 15 00 56" src="https://user-images.githubusercontent.com/13377553/185414033-10e2c0fb-7ecd-4372-bb04-fd4f51e77b31.png">

## How to manually test the feature
- create an application
- go to dashboard
- click delete
- confirm the deletion / cancel the deletion
- return to dashboard - if you chose deletion then the flash message should appear otherwise it will show that the application is still in the list.
